usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+            purposes.append("default")
+        
+        if self.chk_purpose_render:
+            purposes.append("render")
+
+        if self.chk_purpose_proxy:
+            purposes.append("proxy")
+
+        if self.chk_purpose_guide:
+            purposes.append("guide")
+        
+        return purposes
+
+    def _set_total_frames(self, frames: int):
+        """
+        Update progress bar with reported maximum frames
+        """
+        #self._total_frames = frames
+        self.progressbar.setFormat(f"Rendering %v / {frames} frames...")
+        self.progressbar.setMinimum(1)
+        self.progressbar.setMaximum(frames)
+    
+    def _set_render_progress(self, frame: int):
+        """
+        Increment progress bar with reported rendered frames.
+        """
+        #self._render_progress = frame
+        self.progressbar.setValue(frame)
+
+    def _construct_frames_argument(self) -> str:
+        start = self.spinbox_frame_start.value()
+        end = self.spinbox_frame_end.value()
+        stride = self.spinbox_frame_stride.value()
+
+        opt_index = self.cbox_framerange_options.currentIndex()
+
+        if opt_index == 0:
+            return f"{start}"
+        elif opt_index == 1:
+            return playblast.get_frames_string(start,end,stride)  
+        elif opt_index == 2:
+            if self._stageview:
+                return playblast.get_stageview_frame(self._stageview)      
+
+    def _get_camera(self) -> tuple[UsdGeom.Camera,bool]:
+        """
+        Returns a camera and whether it should be destroyed afterwards.
+        -> UsdGeom.Camera, should_destroy()
+        """
+
+        box_text = self.cbox_camera.currentText()
+        data = self.cbox_camera.currentData()
+
+        width = self.spinbox_horresolution.value()
+        height = self.spinbox_verresolution.value()
+
+        aspect_ratio: float = float(width) / float(height)
+
+        if data is not None:
+            data = UsdGeom.Camera(data)
+            v_aperture = data.GetVerticalApertureAttr()
+            # ensure that camera can render desired aspect ratio.
+            if not v_aperture.IsValid():
+                v_aperture = 24
+            else:
+                v_aperture = v_aperture.Get(0)
+
+            data.CreateHorizontalApertureAttr(v_aperture * aspect_ratio)
+            data.CreateHorizontalApertureOffsetAttr(0)
+            data.CreateVerticalApertureAttr(v_aperture)
+            data.CreateVerticalApertureOffsetAttr(0)
+
+            # Camera is in scene, doesn't need to be deleted.
+            return data, False
+
+        elif "Framing" in box_text:
+            camera = framing_camera.create_framing_camera_in_stage(
+                self._stage,
+                name="Playblast_framingCam",
+                fit=self.spinbox_fit.value(),
+                width=width,
+                height=height
+                )
+            return camera, True
+        elif "Viewer" in box_text:
```

Comment:
These should not do a `in` check but just match the full label, no?

Reply:
I will match the full label, I've done something similar in other sections.
This was written when I wasn't yet sure about what to call the label yet, but I knew that it was going to have the keywords "Framing" and "Viewer" in there.
Will change in next push.